### PR TITLE
Widen the test for configure regeneration in Travis

### PR DESCRIPTION
**Pull request by dra27**
`configure` depends on `VERSION` and should be generated whenever the compiler version is bumped.

@damiendoligez - I think it's OK that this configure could be updated in the "last commit"/"first commit" commits, therefore?

With this PR, the Travis branch build of trunk would have (correctly) failed after 4.09 was branched.

I've also fixed an innocuous error message displayed in the `check-typo` job of Travis branch builds - GitHub makes the diff look much more drastic than it is (comment inline)

cc @avsm

### CODE REVIEW

File: tools/ci/travis/travis-ci.sh
Comment (dra27) [inline tools/ci/travis/travis-ci.sh:46]: From here on is identical to the deleted chunk above, just indented.

File: tools/ci/travis/travis-ci.sh
Comment (dra27) [inline tools/ci/travis/travis-ci.sh:40]: There's no FETCH_HEAD for a push, so this just ensure that `TRAVIS_COMMIT` does exist in the repo. Prior to this change, there was an error message from git complaining about the lack of the `FETCH_HEAD` followed by the "TRAVIS_COMMIT exists, so going with it" message.

**Comment by dra27** [comment]
(@avsm - this doesn't resolve the fact that `autogen` doesn't work on macOS, which I'll deal with separately)

**Comment by damiendoligez** [comment]
> @damiendoligez - I think it's OK that this configure could be updated in the "last commit"/"first commit" commits, therefore?

Indeed. Could you add the corresponding instructions in section 5 of `tools/release-checklist`?

**Review by damiendoligez — APPROVED** [review]
Looks good but `autogen` doesn't work on the Release Manager's primary machine... I believe you have some work in progress on that, or should I patch it myself?

**Comment by dra27** [comment]
I have the diagnosis, but not yet a fix for macOS! If you've already got a fix, please go for it! (otherwise I will look at it tomorrow)

**Comment by avsm** [comment]
The very very quick fix for `autogen` is to replace `sed` with `gsed` on macOS. Amazingly, there seems to be no consensus on macOS about how to determine/get GNU sed, so I'm not sure how best to upstream this...

**Comment by dra27** [comment]
Hang on, is the problem here just the `-i`? In which case, the fix is easy and I’ll push it later, as writing `sh` on my phone is alas not easy...

**Comment by avsm** [comment]
No, the extended regexp syntax also fails.  Could you paste an example of the things you are trying to filter out? `grep` might be portable enough for this.

**Comment by dra27** [comment]
OK, that should fix it - `-i` without a backup is not portable, so I've used a temporary file instead. Relative addressing (the `+8`) is an obvious feature, and therefore a GNUism, not Posix, so I've switched to the much more readable `N;N;N;N;N;N;N;N;d;`!

I also added the `-f` flag to `autoconf` to ensure it actually recompiles `configure` even when it thinks its unecessary (#8639 restores the dependency-based rebuilding via `make configure`)

I think this is good to go, therefore?

**Comment by dra27** [comment]
Rebased to account for the change in #8726 - @damiendoligez, are you happy with the changes to `autogen` since your original approval?

**Comment by damiendoligez** [comment]
I'm happy with the autogen changes. I'll let you merge or squash as you prefer.

**Comment by dra27** [comment]
Thanks!

**Pull request merged**
